src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -497,13 +497,12 @@ impl EventProcessor {
                             // When this finally happens, the event will not be synthetic.
                             shared_state_lock.dpi_adjusted = None;
                         } else {
+                            // Unlock shared state to prevent deadlock in callback below
+                            drop(shared_state_lock);
```

Comment:
I think you need to release the lock from other branch as well? Since the release was kind of _fair_?